File: Modeling/app/src/main/java/pisarev/com/modeling/mvp/model/Rect.java
package pisarev.com.modeling.mvp.model;

public class Rect extends Point {
    double height;
    double width;

    public Rect(){}

    public Rect(Point point, float height, float width) {
        x = point.getX();
        z = point.getZ();
        this.height = height;
        this.width = width;
    }

    public void setRect(Point point, float height, float width){
        x = point.getX();
        z = point.getZ();
        this.height = height;
        this.width = width;
    }

    public void setRect(float x,float z, float height, float width){
        this.x = x;
        this.z = z;
        this.height = height;
        this.width = width;
    }

    public boolean isInsideRect(float x, float z) {
        return (x >= this.x && x <= this.x + width) && (z >= this.z && z <= this.z + height);
    }
}

File: Modeling/app/src/main/java/pisarev/com/modeling/mvp/model/Rect.kt
package pisarev.com.modeling.mvp.model

import pisarev.com.modeling.mvp.contur.Point

class Rect {
    var height = 0F
    var width = 0F
    var x = 0F
    var z = 0F

    constructor() {}

    fun setRect(x: Float, z: Float, height: Float, width: Float) {
        this.x = x
        this.z = z
        this.height = height
        this.width = width
    }

    fun isInsideRect(point : Point): Boolean {
        val x = point.x
        val z = point.z
        return x >= this.x && x <= this.x + width && z >= this.z && z <= this.z + height
    }
}